pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participant/ParticipantWrapper.java

```diff
@@ -189,13 +211,27 @@ void fillParticipantWrapperDtosWithProxies(List<ParticipantWrapperDto> result, L
         fieldTypeExtractor.setIndex(esUsersIndex);
         Sort sort = Sort.of(profileCreatedAt, fieldTypeExtractor);
         elasticSearchable.setSortBy(sort);
-        ElasticSearch proxiesByIds = elasticSearchable.getParticipantsByIds(esUsersIndex, proxyGuids);
+        ElasticSearch proxiesByIds = elasticSearchable.getParticipantsByIds(esUsersIndex, proxyGuids, parseParticipantDtos);
         result.forEach(participantWrapperDto -> {
             List<String> participantProxyGuids = participantWrapperDto.getEsData().getProxies();
-            List<ElasticSearchParticipantDto> proxyEsData = proxiesByIds.getEsParticipants().stream()
-                    .filter(elasticSearchParticipantDto -> participantProxyGuids.contains(elasticSearchParticipantDto.getParticipantId()))
-                    .collect(Collectors.toList());
-            participantWrapperDto.setProxyData(proxyEsData);
+            List<ElasticSearchParticipantDto> proxyEsData = null;
+
+            if (parseParticipantDtos) {
+                proxyEsData = proxiesByIds.getEsParticipants().stream()
+                        .filter(elasticSearchParticipantDto -> participantProxyGuids.contains(elasticSearchParticipantDto.getParticipantId()))
+                        .collect(Collectors.toList());
+                participantWrapperDto.setProxyData(proxyEsData);
+            } else {
+                proxyEsData = proxiesByIds.getEsParticipants().stream()
+                        .filter(elasticSearchParticipantDto -> {
+                            Map<String, Object> profile = (Map<String, Object>) elasticSearchParticipantDto.getDataAsMap().get(ESObjectConstants.PROFILE);
```

Comment:
what is the idea behind of filtering by `profile`?

Reply:
profile is where we find the participants GUID

Reply:
that's my main concern of having raw data separately in `ElasticSearchParticipantDto`, it forces to make many casting and verbose code.